src/sentry/reprocessing2.py

```diff
@@ -347,7 +346,75 @@ def _get_info_reprocessed_key(group_id):
     return f"re2:info:{group_id}"
 
 
-def mark_event_reprocessed(data=None, group_id=None, project_id=None):
+def buffered_handle_remaining_events(
+    project_id,
+    old_group_id,
+    new_group_id,
+    datetime_to_event,
+    remaining_events,
+    force_flush_batch=False,
+):
+    """
+    A quick-and-dirty wrapper around `handle_remaining_events` that batches up
+    event IDs in Redis. We need this because Snuba cannot handle many tiny messages and prefers big ones instead.
+
+    Best performance is achieved when timestamps are close together. Luckily we
+    happen to iterate through events ordered by timestamp.
+
+    Ideally we'd have batching implemented via a service like buffers, but for
+    more than counters.
+    """
+
+    client = _get_sync_redis_client()
+    key = f"re2:remaining:{project_id}:{old_group_id}"
+
+    if datetime_to_event:
+        client.lpush(
```

Comment:
It seems that `LPUSH` returns the list size, so we could omit the call to `LLEN` further down.. but only if `LPUSH` is actually called. Not sure if worth it.

https://redis.io/commands/lpush

Reply:
Yup, added.